v2/pkg/genruntime/admissions.go

```diff
@@ -26,3 +28,22 @@ type Defaulter interface {
 	// CustomDefault performs custom defaults that are run in addition to the code generated defaults.
 	CustomDefault()
 }
+
+// ValidateImmutableProperties function validates the update on immutable properties.
+func ValidateImmutableProperties(oldObj MetaObject, newObj MetaObject) error {
+	var errs []error
+
+	if GetResourceIDOrDefault(newObj) == "" {
+		return nil
+	}
+
+	if oldObj.AzureName() != newObj.AzureName() {
+		errs = append(errs, errors.Errorf("updating 'AzureName()' is not allowed for '%s", oldObj.GetName()))
```

Comment:
Why is `AzureName` written as though it's a method call?

```suggestion
		errs = append(errs, errors.Errorf("updating 'AzureName( is not allowed for '%s", oldObj.GetName()))
```

Reply:
Also, does `oldObj.GetName()` return a fully qualified GVK or only a partial name?

Reply:
Just the partial name.

Reply:
We probably dont need version as this error is for all versions always I think? But G + K is good.